tools/iree-prof-output-utils.cc

```diff
@@ -8,75 +8,140 @@
 
 #include "third_party/abseil-cpp/absl/base/log_severity.h"
 #include "third_party/abseil-cpp/absl/flags/parse.h"
-#include "third_party/abseil-cpp/absl/log/check.h"
 #include "third_party/abseil-cpp/absl/log/globals.h"
 #include "third_party/abseil-cpp/absl/log/initialize.h"
+#include "third_party/abseil-cpp/absl/strings/str_cat.h"
 #include "third_party/abseil-cpp/absl/time/clock.h"
 #include "third_party/abseil-cpp/absl/time/time.h"
 #include "third_party/tracy/server/TracyWorker.hpp"
 
 namespace iree_prof {
+namespace {
 
-const char* GetZoneName(const tracy::Worker& worker,
-                        int16_t source_location_id) {
-  const auto& srcloc = worker.GetSourceLocation(source_location_id);
-  return worker.GetString(srcloc.name.active ? srcloc.name : srcloc.function);
+constexpr int kGpuThreadIndicator = 1 << 16;
+
+template <typename T>
+const T& GetEvent(const T& event) { return event; }
+
+template <typename T>
+const T& GetEvent(const tracy::short_ptr<T>& event) {
+  return *event;
 }
 
-void MergeDuration(int64_t start, int64_t end, std::vector<int64_t>& merged) {
-  CHECK_LE(start, end);
-  if (start == end) {
-    return;  // No changes.
+template <typename T>
+int64_t SumDurationInTimeline(const tracy::Worker& worker,
+                              const tracy::Vector<T>& timeline) {
```

Comment:
Please add a comment what `timeline` is.

Reply:
Done